From: Johan Van de Wauw <johan.vandewauw@gmail.com>
Date: Thu, 17 Jun 2010 10:40:42 +0200
Subject: [PATCH] fix issue with grayscale PNGs (patch by Ralf Junker)

Cherry pick from upstream, without changes to the CHANGES file
---
 src/hpdf_image_png.c |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/src/hpdf_image_png.c b/src/hpdf_image_png.c
index d86354b..9a16d4e 100644
--- a/src/hpdf_image_png.c
+++ b/src/hpdf_image_png.c
@@ -572,7 +572,11 @@ no_transparent_color_in_palette:
 		}
 		HPDF_FreeMem(image->mmgr, smask_data);
 
-		ret += HPDF_Dict_AddName (image, "ColorSpace", "DeviceRGB");
+		if (info_ptr->color_type == PNG_COLOR_TYPE_GRAY_ALPHA) {
+			ret += HPDF_Dict_AddName (image, "ColorSpace", "DeviceGray");
+		} else {
+			ret += HPDF_Dict_AddName (image, "ColorSpace", "DeviceRGB");
+		}
 		ret += HPDF_Dict_AddNumber (image, "Width", (HPDF_UINT)info_ptr->width);
 		ret += HPDF_Dict_AddNumber (image, "Height", (HPDF_UINT)info_ptr->height);
 		ret += HPDF_Dict_AddNumber (image, "BitsPerComponent",	(HPDF_UINT)info_ptr->bit_depth);
-- 
